contentful/apps
# feat: add trailing slash configuration option for GA4 [INTEG-1287]

## Purpose

We received feedback that some customers have all of their URLs automatically redirect to a trailing slash. Therefore, when they want to see data in Google Analytics, they would want to see the analytics for `https://mywebsite.com/products/cool-shirt/` instead of `https://mywebsite.com/products/cool-shirt`, as these are tracked as different pages in Google Analytics. Our current GA4 app does not allow for the user to easily configure the app to include a trailing slash.

## Approach

This PR adds a checkbox to the configuration page where users can select if they would like all their URLs to have a trailing slash appended to the end. If this checkbox is selected and the configuration page saved, then in the Sidebar app the user can see that a trailing slash is now added to the end of their page path. This is done via a new installation parameter, `useTrailingSlash`.

**Updated section on configuration screen:**
<img width="856" alt="Screenshot 2023-09-15 at 11 56 55 AM" src="https://github.com/contentful/apps/assets/62958907/5f3e3a2b-d6c1-4a32-ab51-d69274e8485f">

**Updated page path on sidebar app:**
<img width="342" alt="Screenshot 2023-09-15 at 11 52 02 AM" src="https://github.com/contentful/apps/assets/62958907/1755c941-99a5-4400-9f50-3c0eb69c5921">

**Updated page path filter in Google Analytics report:**
<img width="311" alt="Screenshot 2023-09-15 at 11 52 17 AM" src="https://github.com/contentful/apps/assets/62958907/1fc36f69-85fc-4e4a-87f5-e52c577c365d">


## Testing steps

The new checkbox will default to false, so for customers with the app installed, there should be no change.

Check the box on the config page and save the config. The navigate to an entry where the app is enabled in the sidebar. The page path should appear as: `/${urlPrefix}/${slug}/`. When you click on `Open in Google Analytics`, the filter in the report page should reflect this trailing slash as well.

## Breaking Changes

This change should not impact current app installations.

## Dependencies and/or References

## Deployment
